src/sentry/eventstream/kafka/protocol.py

```diff
@@ -104,59 +104,65 @@ def get_task_kwargs_for_message_from_headers(headers: Sequence[Tuple[str, Option
     Same as get_task_kwargs_for_message but gets the required information from
     the kafka message headers.
     """
-    try:
-        header_data = {k: v for k, v in headers}
-        if "group_id" not in header_data:
-            header_data["group_id"] = None
-        if "primary_hash" not in header_data:
-            header_data["primary_hash"] = None
 
-        def decode_str(value: Optional[bytes]) -> str:
-            assert isinstance(value, bytes)
-            return value.decode("utf-8")
+    def decode_str(value: Optional[bytes]) -> str:
+        assert isinstance(value, bytes)
+        return value.decode("utf-8")
 
-        def decode_optional_str(value: Optional[bytes]) -> Optional[str]:
-            if value is None:
-                return None
-            return decode_str(value)
+    def decode_optional_str(value: Optional[bytes]) -> Optional[str]:
+        if value is None:
+            return None
+        return decode_str(value)
 
-        def decode_int(value: Optional[bytes]) -> int:
-            assert isinstance(value, bytes)
-            return int(value)
+    def decode_int(value: Optional[bytes]) -> int:
+        assert isinstance(value, bytes)
+        return int(value)
 
-        def decode_optional_int(value: Optional[bytes]) -> Optional[int]:
-            if value is None:
-                return None
-            return decode_int(value)
+    def decode_optional_int(value: Optional[bytes]) -> Optional[int]:
+        if value is None:
+            return None
+        return decode_int(value)
 
-        def decode_bool(value: bytes) -> bool:
-            return bool(int(decode_str(value)))
+    def decode_bool(value: bytes) -> bool:
+        return bool(int(decode_str(value)))
 
+    try:
+        header_data = {k: v for k, v in headers}
```

Comment:
I think this is what @nikhars was talking about in the meeting today, that we might be building a map of 1000 headers and then only extracting the 10 or so we use below.

Reply:
I see. The only headers that we included in the message are the 10 or so all of which we are using below.